packages/dev/sharedUiComponents/src/lines/optionsLineComponent.tsx

```diff
@@ -80,44 +83,103 @@ export class OptionsLineComponent extends React.Component<IOptionsLineComponentP
+
+        if (isNaN(Number(value))) {
+            for (let i = 0; i < this.props.options.length; i++) {
+                if (this.props.options.at(i)?.label === valueString) {
+                    value = Number(this.props.options.at(i)?.value);
+                }
+            }
+        }
+
+        this.forceUpdate();
+        if (value === 0) {
+            this.setState({ addCustom: true });
+        }
+
         this._localChange = true;
 
         const store = this.props.extractValue ? this.props.extractValue(this.props.target) : this.props.target[this.props.propertyName];
 
         if (!this.props.noDirectUpdate) {
             this.props.target[this.props.propertyName] = this._remapValueOut(value as number);
         }
-        this.setState({ value: value });
-
-        if (this.props.onSelect) {
-            this.props.onSelect(value);
+        if (value != 0) {
+            this.setState({ value: value });
+            if (this.props.onSelect) {
+                this.props.onSelect(value);
+            }
+        } else {
+            if (this.props.onSelect) {
+                this.props.onSelect(this.state.value);
+            }
         }
 
         const newValue = this.props.extractValue ? this.props.extractValue(this.props.target) : this.props.target[this.props.propertyName];
 
         this.raiseOnPropertyChanged(newValue, store);
     }
 
+    updateCustomValue() {
+        this.setState({ addCustom: false });
+    }
+
     render() {
-        return (
-            <div className={"listLine" + (this.props.className ? " " + this.props.className : "")}>
-                {this.props.icon && <img src={this.props.icon} title={this.props.iconLabel} alt={this.props.iconLabel} color="black" className="icon" />}
-                <div className="label" title={this.props.label}>
-                    {this.props.label}
+        if (this.state.addCustom) {
+            return (
+                <div className={"listLine" + (this.props.className ? " " + this.props.className : "")}>
+                    {this.props.icon && <img src={this.props.icon} title={this.props.iconLabel} alt={this.props.iconLabel} color="black" className="icon" />}
+                    <div className="label" title={this.props.label}>
+                        {this.props.label}
+                    </div>
+                    <div className="options">
+                        <input
+                            type="text"
+                            placeholder="Enter a custom font here"
+                            id="customFont"
+                            onKeyDown={(event) => {
+                                event.keyCode === 13 && this.props.addVal != undefined
+                                    ? (this.props.addVal(
+                                          { label: (document.getElementById("customFont") as HTMLInputElement).value, value: this.props.options.length + 1 },

```

Comment:
Nit: Instead of using document.getElementById to get the input, you can just do event.target.value

Reply:
I did try that originally, but I got the error "Property 'value' does not exist on type 'EventTarget'"

Reply:
(event.target as HTMLInputElement).value